src/main/kotlin/pl/edu/agh/gem/internal/service/PaymentService.kt

```diff
@@ -85,10 +87,26 @@ class PaymentService(
             statusHistory = updatedStatusHistory,
         )
     }
+
+    fun deletePayment(paymentId: String, groupId: String, userId: String) {
+        val paymentToDelete = paymentRepository.findByPaymentIdAndGroupId(paymentId, groupId) ?: throw MissingPaymentException(paymentId, groupId)
+
+        if (!userId.isCreator(paymentToDelete)) {
+            throw PaymentDeletionAccessException(userId, paymentId)
+        }
+
+        paymentRepository.delete(paymentToDelete)
+        archivedPaymentRepository.add(paymentToDelete)
```

Comment:
should we just archive payments that was in statuses other that PENDING

Reply:
yes

Reply:
???

Reply:
all expenses and pyaments should be deleteable. If not we should also disable editing accepted expense/payment cause then if u want to delete u just edit it then status is pending and deletion is enabled

Reply:
So in situation when you make an error in creating expense (for example forgot about one of the participants) and other accept it (so status is accepted) you can not change it or delete it

Reply:
i don't see any cons of delete accepted expense/payment. It is decision of creator and only the creator would not get their money back)